runtime/jsMain/src/kotlinx/serialization/DynamicObjectSerializer.kt

```diff
@@ -0,0 +1,252 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.builtins.AbstractEncoder
+import kotlinx.serialization.builtins.list
+import kotlinx.serialization.builtins.nullable
+import kotlinx.serialization.json.JsonConfiguration
+import kotlinx.serialization.json.internal.BEGIN_LIST
+import kotlinx.serialization.json.internal.BEGIN_OBJ
+import kotlinx.serialization.json.internal.END_LIST
+import kotlinx.serialization.json.internal.END_OBJ
+import kotlinx.serialization.modules.EmptyModule
+import kotlinx.serialization.modules.SerialModule
+import kotlinx.serialization.modules.getContextualOrDefault
+import kotlin.math.abs
+import kotlin.math.floor
+
+
+/**
+ * Converts Kotlin data structures to plain Javascript objects
+ *
+ *
+ * Limitations:
+ * * Map keys must be of primitive or enum type
+ * * Enums are serialized as the value of `@SerialName` if present or their name, in that order.
+ * * Currently does not support polymorphism
+ *
+ * Example of usage:
+ * ```
+ *  @Serializable
+ *  open class DataWrapper(open val s: String, val d: String?)
+ *
+ *  val wrapper = DataWrapper("foo", "bar")
+ *  val plainJS: dynamic = DynamicObjectSerializer().serialize(DataWrapper.serializer(), wrapper)
+ * ```
+ *
+ * @param encodeNullAsUndefined if true null properties will be omitted from the output
```

Comment:
This param doc is obsolete now

Reply:
The parameter is still there as well. Is there a well suited parameter that's already part of JsonConfiguration we could use?

Reply:
Indeed, my bad. There is no suitable parameter [yet], so you may leave it as is.